Dockerfile

```diff
@@ -0,0 +1,97 @@
+# added platform flag here because -- https://stackoverflow.com/questions/71040681/qemu-x86-64-could-not-open-lib64-ld-linux-x86-64-so-2-no-such-file-or-direc
+FROM --platform=linux/amd64 python:3.8
+
+WORKDIR /app/oppia
+
+ENV oppia_is_dockerized="true"
+
+
+# installing the pre-requisites libs and dependencies
+RUN apt-get update -y && apt-get upgrade -y \
+    curl \
+    git \
+    python3-dev \
+    python3-setuptools \
+    python3-pip \
+    unzip \
+    python3-yaml \
+    python3-matplotlib \
+    chromium
+
+RUN pip install --upgrade pip==21.2.3
+RUN pip install pip-tools==6.6.2 setuptools==58.5.3 cmake
+
+
+# installing python dependencies from the requirements.txt file
+COPY requirements_dev.in .
+COPY requirements_dev.txt .
+
+RUN pip-compile --generate-hashes requirements_dev.in
+RUN pip install --require-hashes --no-deps -r requirements_dev.txt
+
+
+## installing packages from the package.json file
+COPY package.json .
+COPY scripts/linters/custom_eslint_checks ./scripts/linters/custom_eslint_checks
+RUN apt-get -y install npm
+
+RUN curl -fsSL https://deb.nodesource.com/setup_16.x | bash -
+RUN apt-get install -y nodejs
+
+RUN npm install -g yarn
+RUN yarn install
+
+
+# installing third party dependencies
+COPY scripts ./scripts
+COPY /core ./core
+COPY /assets ./assets
+COPY dependencies.json .
+COPY buf.gen.yaml .
+
+RUN python -m scripts.install_third_party
+RUN python -m scripts.build
```

Comment:
I don't think it's a good idea to run python scripts here for setup, we won't be able to use docker cache functionality properly. Will it recognize changes in build.py file or the imports made in the build.py file? I would highly suggest not using a complicated Python script for Dockerfile. If Docker is providing us a stable env why can't we write simple steps to build the container? Is there any complication?